server/src/extent_tree.c

```diff
@@ -317,18 +317,21 @@ int extent_tree_truncate(
     struct extent_tree* tree, /* tree to truncate */
     unsigned long size)       /* size to truncate extents to */
 {
+    unsigned long target_offset = size - 1;
```

Comment:
the caller may pass in size==0, which would cause this to be (unsigned long)-1.  Rather than deal with that confusion, can we update the checks to consider that size is one more than the logical file offset?